File: AndreasH-julia/Day5/part2.jl
using Printf
function extractBoardingId(boardingPass)
    return parse(Int,replace(replace(replace(replace(boardingPass,'F'=>"0"),'B'=>"1"),'L'=>"0"),'R'=>"1");base=2)
end
rowIDs = []
data = readlines(open("input.txt","r"))
for line in data
    id = extractBoardingId(line)
    push!(rowIDs,id)
end
# If we find where the difference is 2, then we found our boarding pass
for (index,line) in enumerate(sort!(rowIDs))
    if (rowIDs[index+1] - line) == 2
        @printf("Your boarding pass ID is : %d",line+1)
        break
    end
end

File: nattfalk-python/day6/part2.py
group_answers = []

def get_unique_answers():
    everyones_answer = []
    for idx, ga in enumerate(group_answers):
        if idx > 0:
            everyones_answer = list(set(everyones_answer) & set(ga))
        else:
            everyones_answer = list(ga)
    return len(everyones_answer)

answered_questions = 0
with open('input.txt', 'r') as f:
    for line in f:
        line = line.strip()
        if line:
            group_answers.append(list(line))
        else:
            answered_questions += get_unique_answers()
            group_answers = []

if len(group_answers) > 0:
    answered_questions += get_unique_answers()

print(f'Total {answered_questions}')
